eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/sorted/mutable/SynchronizedSortedBag.java

```diff
@@ -655,4 +656,22 @@ public int detectLastIndex(Predicate<? super T> predicate)
     {
         throw new UnsupportedOperationException("asParallel() method is not supported for " + this.getClass().getSimpleName() + '.');
     }
+
+    @Override
+    public Optional<T> getFirstOptional()
+    {
+        synchronized (this.getLock())
+        {
+            return this.getDelegate().getFirstOptional();
+        }
+    }
+
+    @Override
+    public Optional<T> getLastOptional()
+    {
+        synchronized (this.getLock())
+        {
+            return this.getDelegate().getLastOptional();
+        }
+    }
```

Comment:
Ah yes, that is a good point. I think AbstractSynchronizedOrderedIterable is a good idea, but it can come later.